main/autogen/test/constant_hash_test.cc

```diff
@@ -0,0 +1,185 @@
+#include "doctest.h"
+// has to go first as it violates our requirements
+#include "core/ErrorQueue.h"
+#include "core/Unfreeze.h"
+#include "core/core.h"
+#include "main/autogen/constant_hash.h"
+#include "parser/parser.h"
+#include "spdlog/fmt/fmt.h"
+#include "spdlog/sinks/stdout_color_sinks.h"
+#include "spdlog/spdlog.h"
+
+namespace spd = spdlog;
+using namespace std;
+
+namespace sorbet {
+auto logger = spd::stderr_color_mt("parser_test");
+auto errorQueue = make_shared<sorbet::core::ErrorQueue>(*logger, *logger);
+
+struct Helper {
+    core::GlobalState gs;
+    int unique;
+
+    Helper() : gs(core::GlobalState(errorQueue)), unique(0) {
+        gs.initEmpty();
+    }
+
+    unsigned int hashExample(string_view example) {
+        string filename = fmt::format("{}.rb", unique);
+        unique += 1;
+        core::FileRef file;
+        {
+            core::UnfreezeFileTable fileTableAccess(gs);
+            file = gs.enterFile(filename, example);
+        }
+        core::UnfreezeNameTable nameTableAccess(gs);
+        auto settings = parser::Parser::Settings{false, false, false};
+        auto node = parser::Parser::run(gs, file, settings);
+        return autogen::constantHashNode(gs, node.get());
+    }
+};
+
+TEST_CASE("SimpleClass") { // NOLINT
+    Helper helper;
+
+    // we'll use this simple file as our baseline
+    auto basicHash = helper.hashExample("class Foo\n"
```

Comment:
What do you think about hard-coding some of the expected values in this test to ensure hash stability?

Reply:
Hard-coded some examples in the test, too.